In [1]:
# Import lib
import pandas as pd
import numpy as np

In [2]:
csv_file = './data/Census_Data_-_Selected_socioeconomic_indicators_in_Chicago__2008___2012.csv'

df = pd.read_csv(csv_file)

csv_file = './data/Chicago_Public_Schools_-_Progress_Report_Cards__2011-2012_.csv'

df1 = pd.read_csv(csv_file)

csv_file = './data/Crimes_-_2001_to_Present.csv'

df2 = pd.read_csv(csv_file)

In [3]:
# Problem 1
pr1 = df2

print(f"Total crime records: {pr1.shape[0]}")

Total crime records: 7807846


In [4]:
# Problem 2
pr2 = df[["Community Area Number", "COMMUNITY AREA NAME", "PER CAPITA INCOME "]]

pr2=pr2.dropna(how="all")
for index, row in pr2.iterrows():
    if int(row["PER CAPITA INCOME "]) < 11000:
        print(row)

Community Area Number                  26.0
COMMUNITY AREA NAME      West Garfield Park
PER CAPITA INCOME                     10934
Name: 25, dtype: object
Community Area Number              30.0
COMMUNITY AREA NAME      South Lawndale
PER CAPITA INCOME                 10402
Name: 29, dtype: object
Community Area Number           37.0
COMMUNITY AREA NAME      Fuller Park
PER CAPITA INCOME              10432
Name: 36, dtype: object
Community Area Number         54.0
COMMUNITY AREA NAME      Riverdale
PER CAPITA INCOME             8201
Name: 53, dtype: object


In [15]:
#problem 6
#drop NA
df2 = df2.dropna(subset=["Location Description"])

#dữ liệu có nhiều loại school nên tìm các Location Description có chứa SCHOOL
crimes_at_schools = df2[df2["Location Description"].str.contains("SCHOOL", case=False)]["Primary Type"].unique()

print("kind of crimes were recorded at schools:")
for crime in crimes_at_schools:
    print(crime)

kind of crimes were recorded at schools:
BATTERY
ASSAULT
OFFENSE INVOLVING CHILDREN
THEFT
CRIMINAL TRESPASS
MOTOR VEHICLE THEFT
OTHER OFFENSE
CRIMINAL DAMAGE
BURGLARY
WEAPONS VIOLATION
PUBLIC PEACE VIOLATION
CRIMINAL SEXUAL ASSAULT
SEX OFFENSE
LIQUOR LAW VIOLATION
NARCOTICS
STALKING
DECEPTIVE PRACTICE
ROBBERY
INTIMIDATION
HOMICIDE
CONCEALED CARRY LICENSE VIOLATION
ARSON
OBSCENITY
KIDNAPPING
OTHER NARCOTIC VIOLATION
INTERFERENCE WITH PUBLIC OFFICER
CRIM SEXUAL ASSAULT
PUBLIC INDECENCY
GAMBLING
PROSTITUTION
HUMAN TRAFFICKING
RITUALISM


In [5]:
# Problem 6
pr6 = df1[["Safety Score", "Elementary, Middle, or High School"]]

schoolTypes = {}

pr6 = pr6.dropna(how="any")

for index, row in pr6.iterrows():
    try:
        schoolTypes[row["Elementary, Middle, or High School"]] += int(row["Safety Score"])
    except KeyError as e:
        schoolTypes[row["Elementary, Middle, or High School"]] = int(row["Safety Score"])
    except ValueError as e:
        print(e)

for schoolType in schoolTypes:
    print(f"{schoolType}: {schoolTypes.get(schoolType)}")

ES: 20650
HS: 4218
MS: 528


In [14]:
#problem 7
sorted_data = df.sort_values(by="PERCENT HOUSEHOLDS BELOW POVERTY", ascending=False)

# Trích xuất 5 khu vực cộng đồng có tỷ lệ hộ gia đình dưới mức nghèo nhất
top_5_community_areas = sorted_data.head(5)
print("5 community areas with highest % of households below poverty line:")
print(top_5_community_areas[["COMMUNITY AREA NAME", "PERCENT HOUSEHOLDS BELOW POVERTY"]])

5 community areas with highest % of households below poverty line:
   COMMUNITY AREA NAME  PERCENT HOUSEHOLDS BELOW POVERTY
53           Riverdale                              56.5
36         Fuller Park                              51.2
67           Englewood                              46.6
28      North Lawndale                              43.1
26  East Garfield Park                              42.4


# Trả lời câu hỏi

Problem 1: Find the total number of crimes recorded in the CRIME table:
    Total crime records: 7807846
    
Problem 2: List community areas with per capita income less than 11000.
     West Garfield Park
     South Lawndale
     Fuller Park
     Riverdale
     
Problem 3: List all case numbers for crimes involving minors?

Problem 4: List all kidnapping crimes involving a child?(children are not considered minors for the purposes of crime analysis)

Problem 5: What kind of crimes were recorded at schools?
    BATTERY
    ASSAULT
    OFFENSE INVOLVING CHILDREN
    THEFT
    CRIMINAL TRESPASS
    MOTOR VEHICLE THEFT
    OTHER OFFENSE
    CRIMINAL DAMAGE
    BURGLARY
    WEAPONS VIOLATION
    PUBLIC PEACE VIOLATION
    CRIMINAL SEXUAL ASSAULT
    SEX OFFENSE
    LIQUOR LAW VIOLATION
    NARCOTICS
    STALKING
    DECEPTIVE PRACTICE
    ROBBERY
    INTIMIDATION
    HOMICIDE
    CONCEALED CARRY LICENSE VIOLATION
    ARSON
    OBSCENITY
    KIDNAPPING
    OTHER NARCOTIC VIOLATION
    INTERFERENCE WITH PUBLIC OFFICER
    CRIM SEXUAL ASSAULT
    PUBLIC INDECENCY
    GAMBLING
    PROSTITUTION
    HUMAN TRAFFICKING
    RITUALISM
    
Problem 6: List the average safety score for all types of schools!
    ES: 20650
    HS: 4218
    MS: 528
    
Problem 7: List 5 community areas with highest % of households below poverty line.
    Riverdale
    Fuller Park
    Englewood
    North Lawndale
    East Garfield Park   

Problem 8: Which community area(number) is most crime prone?

Problem 9: Use a sub-query to find the name of the community area with highest hardship index.

Problem 10: Use a sub-query to determine the Community Area Name with most number of crimes?
